saltgui/static/scripts/api.js

```diff
@@ -33,12 +42,26 @@ class API {
       } );
     document.querySelector(".run-command input[type='submit']")
       .addEventListener('click', this._onRun);
+
+    // keydown is too early, keypress also does not work
+    document.querySelector("#command")
+      .addEventListener('keyup', this.menu.verifyAll);
+    // cut/paste do not work everywhere
+    document.querySelector("#command")
+      .addEventListener('cut', this.menu.verifyAll);
+    document.querySelector("#command")
+      .addEventListener('paste', this.menu.verifyAll);
+    // blur/focus should not be needed but are a valueable fallback
+    document.querySelector("#command")
+      .addEventListener('blur', this.menu.verifyAll);
+    document.querySelector("#command")
+      .addEventListener('focus', this.menu.verifyAll);
```

Comment:
Would it be nice if tthese specific events would be defined in the Document class and that you just call:
```javascript
Document._registerEventListeners();
``` 

?

Reply:
I want to keep the menu creation in `API.js` because the menu may someday contain items that are not documentation related.
I want to keep the event registration to `menu.verifyAll` in `API.js` because that code is not documentation related. i.e. it has no technical connection to the documentation logic.
Did move the registration of the menu item to the `Documentation` class.